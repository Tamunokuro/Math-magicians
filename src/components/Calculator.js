import React, { useState } from 'react';

const Calculator = () => {
  const [state] = useState({ total: 0 });
  const { total } = state;

  return (
    <section className="calculator-section">
      <div className="calculator">
        <span className="results">
          {' '}
          {total}
          {' '}
        </span>
        <button type="button" className="operators">AC</button>
        <button type="button" className="operators">+/-</button>
        <button type="button" className="operators">%</button>
        <button type="button" className="operators">/</button>
        <button type="button" className="numbers">0</button>
        <button type="button" className="numbers">7</button>
        <button type="button" className="numbers">8</button>
        <button type="button" className="numbers">9</button>
        <button type="button" className="numbers">x</button>
        <button type="button" className="numbers">4</button>
        <button type="button" className="numbers">5</button>
        <button type="button" className="numbers">6</button>
        <button type="button" className="operators">-</button>
        <button type="button" className="numbers">1</button>
        <button type="button" className="numbers">2</button>
        <button type="button" className="numbers">3</button>
        <button type="button" className="operators">+</button>
        <button type="button" className="numbers">0</button>
        <button type="button" className="operators">.</button>
        <button type="button" className="operators">=</button>
      </div>
    </section>
  );
};

export default Calculator;
