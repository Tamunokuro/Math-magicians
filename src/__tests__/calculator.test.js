import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Test for calculator component', () => {
  it('it should render the number of buttons in the calculator', () => {
    render(<Calculator />);
    const buttonElements = screen.getAllByRole('button');
    expect(buttonElements.length).toBe(19);
  });

  it('Snapshot should match component render', () => {
    const calculator = render(<Calculator />);
    expect(calculator).toMatchSnapshot();
  });

  it('it add 7 and 3 By click button', () => {
    render(<Calculator />);
    const getFirtBtn = screen.getByTestId('7');
    const getSecondBtn = screen.getByTestId('3');
    const getOpBtn = screen.getByTestId('+');
    const getEqualBtn = screen.getByTestId('=');
    const calScreen = screen.getByTestId('screen');
    fireEvent.click(getFirtBtn, { target: { innerText: '7' } });
    fireEvent.click(getOpBtn, { target: { innerText: '+' } });
    fireEvent.click(getSecondBtn, { target: { innerText: '3' } });
    fireEvent.click(getEqualBtn, { target: { innerText: '=' } });
    expect(calScreen.textContent).toBe('10');
  });
});
