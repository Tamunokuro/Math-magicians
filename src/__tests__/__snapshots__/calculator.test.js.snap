// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for calculator component Snapshot should match component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <section
          class="calculator-section"
        >
          <div
            class="justify-content-center row"
          >
            <div
              class="col-6"
            >
              <p>
                Let's do some Maths!
              </p>
            </div>
            <div
              class="col-6"
            >
              <div
                class="calculator"
              >
                <span
                  class="results"
                >
                  0
                   
                   
                </span>
                <button
                  class="numbers"
                  name="AC"
                  type="button"
                >
                  AC
                </button>
                <button
                  class="numbers"
                  name="+/-"
                  type="button"
                >
                  +/-
                </button>
                <button
                  class="numbers"
                  name="%"
                  type="button"
                >
                  %
                </button>
                <button
                  class="operators"
                  name="รท"
                  type="button"
                >
                  รท
                </button>
                <button
                  class="numbers"
                  name="7"
                  type="button"
                >
                  7
                </button>
                <button
                  class="numbers"
                  name="8"
                  type="button"
                >
                  8
                </button>
                <button
                  class="numbers"
                  name="9"
                  type="button"
                >
                  9
                </button>
                <button
                  class="operators"
                  name="x"
                  type="button"
                >
                  x
                </button>
                <button
                  class="numbers"
                  name="4"
                  type="button"
                >
                  4
                </button>
                <button
                  class="numbers"
                  name="5"
                  type="button"
                >
                  5
                </button>
                <button
                  class="numbers"
                  name="6"
                  type="button"
                >
                  6
                </button>
                <button
                  class="operators"
                  name="-"
                  type="button"
                >
                  -
                </button>
                <button
                  class="numbers"
                  name="1"
                  type="button"
                >
                  1
                </button>
                <button
                  class="numbers"
                  name="2"
                  type="button"
                >
                  2
                </button>
                <button
                  class="numbers"
                  name="3"
                  type="button"
                >
                  3
                </button>
                <button
                  class="operators"
                  name="+"
                  type="button"
                >
                  +
                </button>
                <button
                  class="numbers span-zero"
                  name="0"
                  type="button"
                >
                  0
                </button>
                <button
                  class="numbers"
                  name="."
                  type="button"
                >
                  .
                </button>
                <button
                  class="operators"
                  name="="
                  type="button"
                >
                  =
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <section
        class="calculator-section"
      >
        <div
          class="justify-content-center row"
        >
          <div
            class="col-6"
          >
            <p>
              Let's do some Maths!
            </p>
          </div>
          <div
            class="col-6"
          >
            <div
              class="calculator"
            >
              <span
                class="results"
              >
                0
                 
                 
              </span>
              <button
                class="numbers"
                name="AC"
                type="button"
              >
                AC
              </button>
              <button
                class="numbers"
                name="+/-"
                type="button"
              >
                +/-
              </button>
              <button
                class="numbers"
                name="%"
                type="button"
              >
                %
              </button>
              <button
                class="operators"
                name="รท"
                type="button"
              >
                รท
              </button>
              <button
                class="numbers"
                name="7"
                type="button"
              >
                7
              </button>
              <button
                class="numbers"
                name="8"
                type="button"
              >
                8
              </button>
              <button
                class="numbers"
                name="9"
                type="button"
              >
                9
              </button>
              <button
                class="operators"
                name="x"
                type="button"
              >
                x
              </button>
              <button
                class="numbers"
                name="4"
                type="button"
              >
                4
              </button>
              <button
                class="numbers"
                name="5"
                type="button"
              >
                5
              </button>
              <button
                class="numbers"
                name="6"
                type="button"
              >
                6
              </button>
              <button
                class="operators"
                name="-"
                type="button"
              >
                -
              </button>
              <button
                class="numbers"
                name="1"
                type="button"
              >
                1
              </button>
              <button
                class="numbers"
                name="2"
                type="button"
              >
                2
              </button>
              <button
                class="numbers"
                name="3"
                type="button"
              >
                3
              </button>
              <button
                class="operators"
                name="+"
                type="button"
              >
                +
              </button>
              <button
                class="numbers span-zero"
                name="0"
                type="button"
              >
                0
              </button>
              <button
                class="numbers"
                name="."
                type="button"
              >
                .
              </button>
              <button
                class="operators"
                name="="
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
